### PR TITLE
test(grouped-character): refactor to ts

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour
- Rename the `grouped-character.cy.js` to `grouped-character.cy.tsx` file in `./cypress/components/grouped-character/` folder.
- Refactor tests to Typescript.

### Current behaviour

Currently GroupedCharacter component is in JS not in TS.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the `grouped-character.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other *.cy.tsx files have regressed
- [x] Run `npm run type-check` to make sure there is no type issue

### CODE REVIEW

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 1]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
-        cy.checkAccessibility();
-      });
+  it("should have the expected border radius styling", () => {
+    CypressMountWithProviders(<GroupedCharacterComponent />);
+    getElement("input").should("have.css", "border-radius", "4px");
+  });
 
-      it("should pass accessibility tests for Groupcharacter AutoFocus story", () => {
-        CypressMountWithProviders(<stories.AutoFocus />);
+  describe("check accessibility for Groupcharacter component", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:239]: Should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 2]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+    });
 
-      it("should pass accessibility tests for Groupcharacter Disabled story", () => {
-        CypressMountWithProviders(<stories.Disabled />);
+    it.each([
+      SIZE.SMALL,
+      SIZE.MEDIUM,
+      SIZE.LARGE,
+    ] as GroupedCharacterProps["size"][])(
+      "should pass accessibility tests for Groupcharacter Sizes %s size story",
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:251]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 3]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+        CypressMountWithProviders(<GroupedCharacterComponent size={size} />);
 
         cy.checkAccessibility();
-      });
+      }
+    );
 
-      it("should pass accessibility tests for Groupcharacter LabelInline story", () => {
-        CypressMountWithProviders(<stories.LabelInline />);
+    it("should pass accessibility tests for Groupcharacter AutoFocus story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:259]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 4]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.AutoFocus />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter LabelInputWidth story", () => {
-        CypressMountWithProviders(<stories.LabelInputWidth />);
+    it("should pass accessibility tests for Groupcharacter Disabled story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:265]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 5]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.Disabled />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter FieldHelp story", () => {
-        CypressMountWithProviders(<stories.FieldHelp />);
+    it("should pass accessibility tests for Groupcharacter LabelInline story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:271]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 6]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.LabelInline />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter LabelHelp story", () => {
-        CypressMountWithProviders(<stories.LabelHelp />);
+    it("should pass accessibility tests for Groupcharacter LabelInputWidth story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:277]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 7]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.LabelInputWidth />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter Required story", () => {
-        CypressMountWithProviders(<stories.Required />);
+    it("should pass accessibility tests for Groupcharacter FieldHelp story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:283]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 8]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.FieldHelp />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter LabelAlign story", () => {
-        CypressMountWithProviders(<stories.LabelAlign />);
+    it("should pass accessibility tests for Groupcharacter LabelHelp story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:289]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 9]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.LabelHelp />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter VariousSeparators story", () => {
-        CypressMountWithProviders(<stories.VariousSeparators />);
+    it("should pass accessibility tests for Groupcharacter Required story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:295]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 10]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter VariousGroups story", () => {
-        CypressMountWithProviders(<stories.VariousGroups />);
+    it.each(["right", "left"] as GroupedCharacterProps["labelAlign"][])(
+      "should pass accessibility tests for Groupcharacter %s LabelAlign story",
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:302]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 11]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+        );
 
         cy.checkAccessibility();
-      });
+      }
+    );
 
-      it("should pass accessibility tests for Groupcharacter ValidationsStringLabel story", () => {
-        CypressMountWithProviders(<stories.ValidationsStringLabel />);
+    it("should pass accessibility tests for Groupcharacter Validations story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:335]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 12]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.Validations />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter ValidationsBoolean story", () => {
-        CypressMountWithProviders(<stories.ValidationsBoolean />);
+    it("should pass accessibility tests for Groupcharacter ValidationsStringComponent story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:341]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 13]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+      CypressMountWithProviders(<stories.ValidationsStringComponent />);
 
-        cy.checkAccessibility();
-      });
+      cy.checkAccessibility();
+    });
 
-      it("should pass accessibility tests for Groupcharacter ValidationsRedesign story", () => {
-        CypressMountWithProviders(<stories.ValidationsRedesign />);
+    it("should pass accessibility tests for Groupcharacter ValidationsStringLabel story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:347]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 14]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
-      // FE-4639
-      describe.skip("should render Groupcharacter component", () => {
-        it("should pass accessibility tests for Groupcharacter ValidationsTooltip story", () => {
-          CypressMountWithProviders(<stories.ValidationsTooltip />);
+      cy.checkAccessibility();
+    });
 
-          cy.checkAccessibility();
-        });
+    it("should pass accessibility tests for Groupcharacter ValidationsBoolean story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:353]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 15]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
 
-        it("should pass accessibility tests for Groupcharacter ValidationsTooltipLabel story", () => {
-          CypressMountWithProviders(<stories.ValidationsTooltipLabel />);
+      cy.checkAccessibility();
+    });
 
-          cy.checkAccessibility();
-        });
-      });
+    it("should pass accessibility tests for Groupcharacter ValidationsRedesign story", () => {
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:359]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 16]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+
+      cy.checkAccessibility();
     });
-  });
 
-  it("should have the expected border radius styling", () => {
-    CypressMountWithProviders(<GroupedCharacterComponent />);
-    getElement("input").should("have.css", "border-radius", "4px");
+    it.each(["error", "warning", "info"])(
+      "should pass accessibility tests for Groupcharacter with %s validation type",
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:366]: As above, should read `GroupedCharacter`.

File: cypress/components/grouped-character/grouped-character.cy.tsx [hunk 17]
```diff
@@ -181,177 +185,206 @@ context("Tests for GroupedCharacter component", () => {
...
+        CypressMountWithProviders(
+          <GroupedCharacterComponent {...{ [validationType]: "Message" }} />
+        );
+
+        cy.checkAccessibility();
+      }
+    );
+
+    it.each(["error", "warning", "info"])(
+      "should pass accessibility tests for Groupcharacter with %s validation type on label",
```
Comment (sianford) [inline cypress/components/grouped-character/grouped-character.cy.tsx:377]: As above, should read `GroupedCharacter`.

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**